### DISCUSSION TITLE
MSB0001: Internal MSBuild Error: while using pac pcf push --publisher-prefix

### DISCUSSION THREAD

**Discussion by IsHafid** [microsoft/powerplatform-build-tools#603] [Q&A]
i ve been using pac pcf push --publisher-prefix dev to push my code in the environement for the past 4 months , and it was working flawlessly ,and today some i m getting this error without doing any changes or updates :

MSB0001: Internal MSBuild Error: The switch name extracted from either the partially or completely unquoted arg should be the same.
i don't understand the error and i can't find a solution to it

**Comment by petrochuk** [comment]
can you share entire command you are using?

**Reply by IsHafid** [reply]
that's the command that i m using : pac pcf push --publisher-prefix dev 
and here is the whole error :
Microsoft.Build.Shared.InternalErrorException: MSB0001: Internal MSBuild Error: The switch 
name extracted from either the partially or completely unquoted arg should be the same.    
   � Microsoft.Build.Shared.ErrorUtilities.ThrowInternalError(String message, Exception innerException, Object[] args)
   � Microsoft.Build.CommandLine.MSBuildApp.ExtractSwitchParameters(String commandLineArg, 
String unquotedCommandLineArg, Int32 doubleQuotesRemovedFromArg, String switchName, Int32 switchParameterIndicator)
   � Microsoft.Build.CommandLine.MSBuildApp.GatherCommandLineSwitches(List`1 commandLineArgs, CommandLineSwitches commandLineSwitches)
   � Microsoft.Build.CommandLine.MSBuildApp.GatherAllSwitches(String commandLine, CommandLineSwitches& switchesFromAutoResponseFile, CommandLineSwitches& switchesNotFromAutoResponseFile)
   � Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)
[ERROR] Exception non g�r�e�: Microsoft.Build.Shared.InternalErrorException: MSB0001: Internal MSBuild Error: The switch name extracted from either the partially or completely unquoted arg should be the same.
[ERROR]    � Microsoft.Build.Shared.ErrorUtilities.ThrowInternalError(String message, Exception innerException, Object[] args)
[ERROR]    � Microsoft.Build.CommandLine.MSBuildApp.ExtractSwitchParameters(String commandLineArg, String unquotedCommandLineArg, Int32 doubleQuotesRemovedFromArg, String switchName, Int32 switchParameterIndicator)
[ERROR]    � Microsoft.Build.CommandLine.MSBuildApp.GatherCommandLineSwitches(List`1 commandLineArgs, CommandLineSwitches commandLineSwitches)
[ERROR]    � Microsoft.Build.CommandLine.MSBuildApp.GatherAllSwitches(String commandLine, CommandLineSwitches& switchesFromAutoResponseFile, CommandLineSwitches& switchesNotFromAutoResponseFile)
[ERROR]    � Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)
[ERROR]    � Microsoft.Build.CommandLine.MSBuildApp.Main()
Microsoft PowerPlatform CLI
Version : 1.28.3+g4ba7dfd

**Comment by kneely** [comment]
Getting the same error. With the same command

**Comment by joem-msft** [comment]
It looks like there may be a breaking change with the msbuild command executable on the machine, and how the pcf push is calling msbuild.

Please open a support ticket where we can ask for logs to verify.

**Comment by RamNemely** [comment]
Yes error for me as well. Worked as expected until Wednesday and started seeing this issue from yesterday. pac pcf push --pp solutionprefix is failing.

**Comment by petrochuk** [comment]
So far, I was not able to repo it. It might be good to open [support ticket](https://learn.microsoft.com/en-us/power-platform/admin/get-help-support) and reference this GitHub issue for faster routing. You can also attach full log files to the support ticket.

**Comment by mattybeard** [comment]
+1 I'm also having this issue - please let me know how I can assist with logs

For now I downgraded to [1.27.6](https://www.nuget.org/packages/Microsoft.PowerApps.CLI/1.27.6) and it's working again.
